cloudinit/sources/__init__.py

```diff
@@ -174,12 +174,20 @@ class DataSource(metaclass=abc.ABCMeta):
 
     # The datasource defines a set of supported EventTypes during which
     # the datasource can react to changes in metadata and regenerate
-    # network configuration on metadata changes.
-    # A datasource which supports writing network config on each system boot
-    # would call update_events['network'].add(EventType.BOOT).
+    # network configuration on metadata changes. These are defined in
+    # `supported_network_events`.
+    # The datasource also defines a set of default EventTypes that the
+    # datasource can react to. These are the event types that will be used
+    # if not overridden by the user.
+    # A datasource requiring to write network config on each system boot
+    # would call default_update_events['network'].add(EventType.BOOT).
 
     # Default: generate network config on new instance id (first boot).
-    update_events = {'network': set([EventType.BOOT_NEW_INSTANCE])}
+    supported_update_events = {'network': set([
+        EventType.BOOT_NEW_INSTANCE,
+        EventType.BOOT,
+    ])}
+    default_update_events = {'network': set([EventType.BOOT_NEW_INSTANCE])}
```

Comment:
Should we use a constant for `'network'` so that we don't have a "magic" string affecting behaviour? (We'll notice `EventScope.NETWROK` a lot sooner than `'netwrok'` via linting, etc.)